recipe/patches/0002-run-ldid-after-install_name_tool.patch

```diff
@@ -1,57 +1,76 @@
-From a489b296b895216167ed6b44833731d448b7b5db Mon Sep 17 00:00:00 2001
+From fea8e31e4029cb1914042c1cf72db3f941928614 Mon Sep 17 00:00:00 2001
 From: Isuru Fernando <isuruf@gmail.com>
-Date: Wed, 30 Sep 2020 21:01:34 -0500
-Subject: [PATCH 2/2] run ldid after install_name_tool
+Date: Fri, 2 Oct 2020 16:02:19 -0500
+Subject: [PATCH] Run ldid after install_name_tool
 
 ---
- cctools/misc/install_name_tool.c | 26 +++++++++++++++++++++++++-
- 1 file changed, 25 insertions(+), 1 deletion(-)
+ cctools/misc/install_name_tool.c | 38 +++++++++++++++++++++++++++++++-
+ 1 file changed, 37 insertions(+), 1 deletion(-)
 
 diff --git a/cctools/misc/install_name_tool.c b/cctools/misc/install_name_tool.c
-index 34d063d..e976ede 100644
+index 34d063d..f4d34e1 100644
 --- a/cctools/misc/install_name_tool.c
 +++ b/cctools/misc/install_name_tool.c
-@@ -131,6 +131,8 @@ char **envp)
+@@ -31,6 +31,7 @@
+ #include "stuff/rnd.h"
+ #include "stuff/allocate.h"
+ #include "stuff/write64.h"
++#include <sys/param.h>
+ 
+ /* used by error routines as the name of the program */
+ char *progname = NULL;
+@@ -131,6 +132,9 @@ char **envp)
      uint32_t narchs;
      char *input;
      char *output;
-+    char *ldidcommand;
-+    int isarm64;
++    char *ldidcommand;  // cctools-port
++    int isarm64, bufsize;        // cctools-port
++    char *p, *prefix, buf[MAXPATHLEN], resolved_name[PATH_MAX];  // cctools-port
  
  	output = NULL;
  	progname = argv[0];
-@@ -312,11 +314,33 @@ char **envp)
+@@ -312,11 +316,43 @@ char **envp)
  	if(errors)
  	    exit(EXIT_FAILURE);
  
-+    isarm64 = 0;
-+    for (i = 0; i < narchs; i++) {
-+        if (archs[i].object->mh != NULL) {
-+            if (archs[i].object->mh->cputype == CPU_TYPE_ARM64) {
-+                isarm64 = 1;
-+            }
-+        } else {
-+            if (archs[i].object->mh64->cputype == CPU_TYPE_ARM64) {
-+                isarm64 = 1;
-+            }
++	isarm64 = 0;  /* cctools-port */
++	for (i = 0; i < narchs; i++) {
++	    if (archs[i].object->mh != NULL) {
++	        if (archs[i].object->mh->cputype == CPU_TYPE_ARM64) {
++	            isarm64 = 1;
++	        }
++	    } else {
++	        if (archs[i].object->mh64->cputype == CPU_TYPE_ARM64) {
++	            isarm64 = 1;
++	        }
 +	    }
-+    }
++	}  /* cctools-port */
 +
  	if(output != NULL)
  	    writeout(archs, narchs, output, 0777, TRUE, FALSE, FALSE, FALSE,
  		     NULL);
 -	else
 +	else {
  	    write_on_input(archs, narchs, input);
-+        output = input;
-+    }
++	    output = input;
++	}
 +
-+    if (isarm64 == 1) {
-+        ldidcommand = allocate(strlen(output) + 9);
-+        memcpy(ldidcommand, "ldid -S ", 8);
-+        memcpy(ldidcommand + 8, input, strlen(output) + 1);
-+        system(ldidcommand);
-+    }
++	if (isarm64 == 1) { /* cctools-port */
++	    bufsize = MAXPATHLEN;
++	    p = buf;
++	    i = _NSGetExecutablePath(p, &bufsize);
++	    if(i == -1){
++		p = allocate(bufsize);
```

Comment:
is the memory attached to p being leaked here?

Reply:
yes. it's a common code pattern used throughout the cctools codebase.

Reply:
kk - I guess these tools return quickly so no harm